### DISCUSSION TITLE
Why do merged routers use `.` and not `/`?

### DISCUSSION THREAD

**Discussion by mmkal** [trpc/trpc#751] [Q&A] [answered]
In the docs (e.g. [merging routers](https://trpc.io/docs/merging-routers)) routers are merged with a `.` prefix:

```ts
const appRouter = createRouter()
  .merge('users.', users) // prefix user procedures with "users."
  .merge('posts.', posts) // prefix poosts procedures with "posts."
```

Is there a reason for not using `/` which is a more standard way of delimiting url path segments? I have a vague memory of seeing an issue somewhere that would actually enforce this too. I ask because we're looking into using [API Gateway with paths mapping](https://aws.amazon.com/about-aws/whats-new/2021/03/amazon-api-gateway-custom-domain-names-support-multi-level-base-path-mappings/) to separate lambdas, all implemented by TRPC. Locally and in the type-system, we can just consider it one API, but each procedure can have granular permissions (read APIs not allowed to do `dynamodb:PutItem`, etc.).

I haven't tried to do API Gateway path mapping with dots yet, but based on the docs it looks not possible. Will we be able to use slashes for this use case? By that I mean

```ts
const appRouter = createRouter()
  .merge('users/', users)
  .merge('posts/', posts)
```

**Comment by KATT** [comment, marked as answer]
You are able to name your tRPC-router `[...trpc].tsx` and use `/` instead. 

I actually had this originally; the reason I changed this myself is that you see the full path in the inspector rather than just the end part.

Note: I have not tested `/`-paths since adding batching support.

**Reply by mmkal** [reply]
Makes sense, thank you! Ok, I will try out slashes and can let you know if we see any issues with batching.

**Reply by KATT** [reply]
Should work - here's the source that should solve it if you're curious: https://github.com/trpc/trpc/blob/019cc9c45a5162f131feac1866e0a5ca77170789/packages/server/src/adapters/next.ts#L30-L32
